gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeUtils.java

```diff
@@ -13,94 +12,51 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import org.hypertrace.gateway.service.entity.config.TimestampConfigs;
-import org.hypertrace.gateway.service.v1.entity.EntitiesRequest;
+import org.hypertrace.gateway.service.common.ExpressionContext;
 
 public class ExecutionTreeUtils {
   /**
    * Returns a non-empty optional if all the attributes in the selection(attributes and
    * aggregations), time aggregations, filter and order by can be read from the same source.
    *
-   * @param executionContext
+   * @param expressionContext
    * @return
    */
-  static Optional<String> getSingleSourceForAllAttributes(ExecutionContext executionContext) {
-    Optional<String> singleSourceFromKeySets = getSingleSourceFromKeySets(executionContext);
+  public static Optional<String> getSingleSourceForAllAttributes(
```

Comment:
Dont remember why we added these methods as utils. Can't these be defined within ExpressionContext?

Reply:
Can move them inside `ExpressionContext` too